src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/TransactionForRpc.cs

```diff
@@ -80,21 +80,20 @@ internal static void RegisterTransactionType<T>() where T : TransactionForRpc, I
 
         public override TransactionForRpc? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            using var document = JsonDocument.ParseValue(ref reader);
-
-            // TODO: For some reason we might get an object wrapped in a String
-            using JsonDocument jsonObject = document.RootElement.ValueKind == JsonValueKind.String
-                ? JsonDocument.Parse(document.RootElement.GetString()!)
-                : document;
-
-            TxType discriminator = DefaultTxType;
-            if (jsonObject.RootElement.TryGetProperty("type", out JsonElement typeProperty))
+            if (reader.ValueIsEscaped)
             {
-                discriminator = (TxType?)typeProperty.Deserialize(typeof(TxType), options) ?? DefaultTxType;
+                string escapedJson = JsonSerializer.Deserialize<string>(ref reader);
+                return JsonSerializer.Deserialize<TransactionForRpc>(escapedJson, options);
             }
 
+            // Copy reader so we can double parse to get type
+            Utf8JsonReader copyReader = reader;
+            TransactionType type = JsonSerializer.Deserialize<TransactionType>(ref copyReader, options);
+
+            TxType discriminator = (TxType)(type.Type ?? (ulong)DefaultTxType);
```

Comment:
Can we instead of deserializing and throwing away result,  just find the discriminator property?

Reply:
Ah it kind of does due to `TransactionType` being a simple class with one field only, maybe change it to struct?

Reply:
Changed `TransactionType` to struct